src/components/znkModuleResults/znkModuleResult.service.js

```diff
@@ -0,0 +1,92 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.znkModuleResults').service('ZnkModuleResultsService', [
+        'InfraConfigSrv', '$log', '$q', 'StorageSrv', 'UtilitySrv',
+        function (InfraConfigSrv, $log, $q, StorageSrv, UtilitySrv) {
+
+            var znkModuleResultsService = {};
+            var MODULE_RESULTS_GUIDS_PATH = StorageSrv.variables.appUserSpacePath + '/moduleResults';
+            var MODULE_RESULTS_PATH = 'moduleResults';
+
+            function _isValidNumber(number){
+                if(!angular.isNumber(number) && !angular.isString(number)){
+                    return false;
+                }
+
+                return !isNaN(+number);
+            }
+
+            function _getModuleResultsGuids(){
+                var storage = InfraConfigSrv.getStorageService();
+                return storage.get(MODULE_RESULTS_GUIDS_PATH);
+            }
+
+            function _getInitModuleResult(moduleId, guid){
+                return $q.when( {
+                    "moduleId": moduleId,
+                    "assign": false,
+                    "contentAssign": false,
+                    "date": null,
```

Comment:
don't sure it's right.
The module result object can be created and saved in firebase, when the teacher asked assign results for a student. in that case I think Its incorrect to to save timestamp. this field will be populate when the user will start to answer the exercise.
what do you think?